### ISSUE TITLE
[BERT/TF] gradients scale up with the number of gradient accumulation

### ISSUE THREAD

**Issue by Oneplus** [NVIDIA/DeepLearningExamples#713]
Related to **BERT/TF** 

**Describe the bug**
This might be an algorimatic bug.

Considering the gradient accumulation code in  https://github.com/NVIDIA/DeepLearningExamples/blob/master/TensorFlow/LanguageModeling/BERT/optimization.py#L112-L164

The final gradient updates happen at L156 and the gradient is a summation of normalized gradients of previous m steps (where m eq the num_gradient_accumulate). Supposing m is very large (it happens when pretraining a BERT-large with large batch size on limited resource), this gradient with scale up with m, and it might harms the training.

To my understanding, the correct way would be accumulating the real grads and moving the gradient clipping into the update() function.

**Comment by swethmandava** [comment]
Hi @Oneplus - Adam (and lamb) are scale-invariant. In our experiments, both clipping before and after accumulating perform similarly. Our [PyTorch version](https://github.com/NVIDIA/DeepLearningExamples/tree/master/PyTorch/LanguageModeling) clips after.
